### ISSUE TITLE
yarn run gulp - Problem finding locally-installed `gulp` 

### ISSUE THREAD

**Issue by doctorx32** [Sylius/PluginSkeleton#74]
After running `yarn install`, is unable to run successfully `yarn run gulp`, which output is:

```
\tests\Application> yarn run gulp
yarn run v0.22.0
$ gulp
[14:31:24] Using gulpfile C:\Coding\php\SyliusMyPlugin\tests\Application\gulpfile.js
[14:31:24] Starting 'admin'...
[14:31:24] Finished 'admin' after 11 ms
[14:31:24] Starting 'shop'...
[14:31:24] Finished 'shop' after 3.82 ms
[14:31:24] Starting 'default'...
[14:31:24] Finished 'default' after 41 μs
[gulp-chug] Gulpfile, Gulpfile.js, contents is empty. Reading directly from disk...

events.js:160
      throw er; // Unhandled 'error' event
      ^
Error: Problem finding locally-installed `gulp` for gulpfile [35mundefined[39m. (Try running `npm install gulp` from [35mC:\Coding\php\SyliusMyPlugin\vendor\sylius\sylius\src\Sylius\Bundle\AdminBundle[39m to install a local gulp for said gulpfile.)

Error: Cannot find module 'gulp' from 'C:\Coding\php\SyliusMyPlugin\vendor\sylius\sylius\src\Sylius\Bundle\AdminBundle'
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Also tried to run `npm install gulp`, which also installed globally, but it not helped. Windows 10 x64

**Comment by stefandoorn** [comment]
Having the same issue here. Was working on syncing `.travis.yml` with the skeleton version, but can't find a solution for this.

https://travis-ci.org/stefandoorn/google-tag-manager-plugin/jobs/334128417

**Comment by stefandoorn** [comment]
@pamil Any clue?

**Comment by pamil** [comment]
@doctorx32 could you try upgrading your skeleton to the most recent version? I've done that on @stefandoorn's bundle and it seems to be already solved in here (https://github.com/stefandoorn/google-tag-manager-plugin/pull/13).

**Comment by doctorx32** [comment]
I tried on mentioned project with `composer update` and on new vanilla Skeleton project, in both cases I received the same error above

**Comment by smoked** [comment]
Anyone solved this problem?

**Comment by smoked** [comment]
I found temporary solution

```bash
ln -s /path/to/tests/Application/node_modules /path/to/vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/
ln -s /path/to/tests/Application/node_modules /path/to/vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle/
```

**Comment by mikemix** [comment]
Same here.

**Comment by Gounlaf** [comment]
@mikemix Make sure to recreate "node_modules" (the one at root of plugin folder) symlink after installing js/nodejs dependencies

**Comment by cefuroX** [comment]
Well I still got the same issue. Linking the mentioned folders didn't helped. 
Is there a solution for this? Can't work like that

**Comment by DragonZoul** [comment]
Well, this ticket started in jannuary and is still open. Any progress after 1 year?

**Comment by oallain** [comment]
I got same error on hold version, but now it's fixe.

**Comment by elcuro** [comment]
I have the similar error on fresh installation after the `yarn build`

`vagrant@ubuntu-bionic:/vagrant/tests/Application$ yarn build
yarn run v1.17.3
$ gulp build
[15:25:54] Requiring external module babel-register
[15:25:54] Using gulpfile /vagrant/tests/Application/gulpfile.babel.js
[15:25:54] Starting 'build'...
[15:25:54] Starting 'buildAdmin'...
[15:25:54] Starting 'buildShop'...
[gulp-chug] Gulpfile, gulpfile.babel.js, contents is empty. Reading directly from disk...
[15:25:54] 'buildAdmin' errored after 25 ms
[15:25:54] Error in plugin 'gulp-chug'
Message:
    Problem finding locally-installed `gulp` for gulpfile undefined. (Try running `npm install gulp` from /vagrant/vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle to install a local gulp for said gulpfile.)

Error: Cannot find module 'gulp' from '/vagrant/vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle'
Details:
    domainEmitter: [object Object]
    domainThrown: false
[15:25:54] 'build' errored after 28 ms
[15:25:54] The following tasks did not complete: buildShop
[15:25:54] Did you forget to signal async completion?
error Command failed with exit code 1.
`

**Comment by lbossut** [comment]
Had the same error on **Windows**, caused by the symlink. (skeleton v1.6.4)
`node_modules` on the rooot directory is a symlink but it's not working on Windows (even if replaced by a Windows symlink), so I basically copy/paste the content of `tests\Application\node_modules` to `node_modules`.

Hope this will help some

**Comment by vvasiloi** [comment]
Workaround:
```
cd tests/Application/
yarn install --modules-folder=../../node_modules/
```

**Comment by pamil** [comment]
@vvasiloi would you be willing to add this workaround to the quick installation guide in the readme?

**Comment by vvasiloi** [comment]
@pamil I will try setting `--nodeModulesPath`, it is a better option if it works.

**Comment by pamil** [comment]
Will it be solved by #186? Or is it a different problem?

**Comment by vvasiloi** [comment]
@pamil yes, it will. The workaround I was talking about works only after removing the symlink.
I ended up using Docker containers, because I can use different versions and it's more convenient:
`docker run -it --rm -v "$PWD":/usr/src/app -w /usr/src/app --name node node:6-alpine yarn install`
